### PR TITLE
SPM archive/upload work-arounds

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

This PR addresses two issues we encountered when archiving and uploading the SPMTest demo app to Test Flight.

1) Cardinal code-signing issue

In the [Cardinal FAQ](https://cardinaldocs.atlassian.net/wiki/spaces/CMSDK/pages/856064035/iOS+Frequently+Asked+Questions), they mention a run script that's required in order to upload apps. It removes simulator architectures from CardinalMobile.framework. We are giving merchants 2 options -- either add this run script OR manually run a command to remove simulator slices prior to archiving.

2) Extra copy of Kount static library

Kount's xcframework wraps their static library (`libKountDataCollector.a`), and there's a [known bug](https://forums.swift.org/t/packaging-static-library-in-spm-package-for-ios-executable/41245/13) where static libraries inside xcframeworks are copied during the build phase. The work-around is to edit your scheme and add a post-action run script that removes the extra copy. (We found a [similar issue and workaround on the Firebase GitHub page](https://github.com/firebase/firebase-ios-sdk/issues/6472#issuecomment-694449182).)

Additional changes to SPMTest app:
- Move CardinalMobile.framework into a better location
- Add app icon in order to upload to Test Flight
- Add `NSLocationWhenInUseUsageDescription` to plist in order to upload to Test Flight

### Checklist

- ~Added a changelog entry~

- @scannillo 
- @sestevens

### CODE REVIEW

File: SampleApps/SPMTest/SPMTest.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -214,6 +205,26 @@
...
+			);
+			inputPaths = (
+			);
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "# Type a script or drag a script file from your workspace to insert its path.\n\nFRAMEWORK=\"CardinalMobile\"\n#FRAMEWORK_EXECUTABLE_PATH is the path where Cardinal framework is located, check Cardinal framework path and update accordingly\nFRAMEWORK_EXECUTABLE_PATH=\"${BUILT_PRODUCTS_DIR}/${FRAMEWORKS_FOLDER_PATH}/$FRAMEWORK.framework/$FRAMEWORK\"\nEXTRACTED_ARCHS=()\nfor ARCH in $ARCHS\ndo\nlipo -extract \"$ARCH\" \"$FRAMEWORK_EXECUTABLE_PATH\" -o \"$FRAMEWORK_EXECUTABLE_PATH-$ARCH\"\nEXTRACTED_ARCHS+=(\"$FRAMEWORK_EXECUTABLE_PATH-$ARCH\")\ndone\nlipo -o \"$FRAMEWORK_EXECUTABLE_PATH-merged\" -create \"${EXTRACTED_ARCHS[@]}\"\nrm \"${EXTRACTED_ARCHS[@]}\"\nrm \"$FRAMEWORK_EXECUTABLE_PATH\"\nmv \"$FRAMEWORK_EXECUTABLE_PATH-merged\" \"$FRAMEWORK_EXECUTABLE_PATH\"\n";
```
Comment (sestevens) [inline SampleApps/SPMTest/SPMTest.xcodeproj/project.pbxproj:224]: This is the run script to remove simulator architectures from CardinalMobile.framework in our SPMTest sample app.

File: SampleApps/SPMTest/SPMTest.xcodeproj/xcshareddata/xcschemes/SPMTest.xcscheme [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
+   version = "1.7">
    <BuildAction
       parallelizeBuildables = "YES"
       buildImplicitDependencies = "YES">
+      <PostActions>
+         <ExecutionAction
+            ActionType = "Xcode.IDEStandardExecutionActionsCore.ExecutionActionType.ShellScriptAction">
+            <ActionContent
+               title = "Run Script"
+               scriptText = "# Type a script or drag a script file from your workspace to insert its path.&#10;rm -rf &quot;${TARGET_BUILD_DIR}/${TARGET_NAME}.app/Frameworks/libKountDataCollector.a&quot;&#10;">
```
Comment (sestevens) [inline SampleApps/SPMTest/SPMTest.xcodeproj/xcshareddata/xcschemes/SPMTest.xcscheme:13]: This is the post-action script to remove the extra copy of `libKountDataCollector.a`.

**Review by sshropshire — APPROVED** [review]

File: SWIFT_PACKAGE_MANAGER.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Support for Swift Package Manager was introduced in Braintree iOS v5.
+
+* [General Instructions](#general-instructions)
+* [Binary Dependencies](#binary-dependencies)
+* [BraintreeDataCollector](#braintreedatacollector)
+* [BraintreeThreeDSecure](#braintreethreedsecure)
+
+### General Instructions
+
+To add the `Braintree` package to your Xcode project, select File > Swift Packages > Add Package Dependency and enter `https://github.com/braintree/braintree_ios` as the repository URL. Tick the checkboxes for the specific Braintree libraries you wish to include.
```
Comment (sarahkoop) [inline SWIFT_PACKAGE_MANAGER.md:12]: Nitpick, but it looks like there is some inconsistency between italics, quotes, and plain text to note selections or buttons to click (for example this line vs line 31 vs line 57)

Comment (scannillo) [reply SWIFT_PACKAGE_MANAGER.md:12]: Got it - will clean this up now

**Review by sarahkoop — APPROVED** [review]
🌟

**Pull request merged**